node/cn/api_backend.go

```diff
@@ -75,8 +75,10 @@ func (b *CNAPIBackend) CurrentBlock() *types.Block {
 }
 
 func (b *CNAPIBackend) SetHead(number uint64) {
-	//b.cn.protocolManager.downloader.Cancel()
+	b.cn.protocolManager.Downloader().Cancel()
+	b.cn.protocolManager.SetTmpStop(true)
 	b.cn.blockchain.SetHead(number)
+	b.cn.protocolManager.SetTmpStop(false)
```

Comment:
1. There is no way to call CN's protocolmanager in methods of `blockchain` package.
2. blockchain.SetHead is called only in two functions except API call. 
-  `NewBlockChain` is an initialize function so we don't need to care about peer sync.
- `ResetWithGenesisBlock`is called only in `loadLastState()` when Head block is missing or got some problem with db. And `loadLastState()` is used when making a new blockchain or reset the canonical blocks. Kind of initializing process. 

so I think in case of non-api function calls of `setHead`, we don't need to care about sync stop process.